### ISSUE TITLE
Do not consider lazy-imports when finding inline target AST nodes when injecting dependencies to shell

### ISSUE THREAD

**Issue by usergenic** [Polymer/polymer-bundler#534]
https://github.com/Polymer/polymer-bundler/pull/528 introduced a bug that basically shoves inlined html dependencies inside dom-modules when a lazy-import inside a dom-module is the first identified import that has a dependency on the injected import, (injected means the import is part of the bundle because the strategy put it there, not because it was in the transitive graph of the bundle entrypoint (typically happens in shell merge).

So anyways you get this kind of chaos with polymer-starter-kit:
```html
<dom-module id="my-app">
  <!--
@license
Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
Code distributed by Google as part of the polymer project is also
subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
--><!-- shared styles for all views -->
<dom-module id="shared-styles">
  <template>
    <style>
      .card {
...
      }
    </style>
  </template>
</dom-module>
<link rel="lazy-import" group="view1" href="my-view1.html">
  <link rel="lazy-import" group="view2" href="my-view2.html">
  <link rel="lazy-import" group="view3" href="my-view3.html">
  <link rel="lazy-import" group="view404" href="my-view404.html">
```

The injection target logic is sound except that it currently doesn't discriminate between import and lazy-import when finding the target to prepend the injected import link to.

As a secondary concern, we may want to be extra careful to not choose injection points inside certain container elements like dom-module.

**Comment by usergenic** [comment]
@bubbit @stramel @elvomka

**Comment by stramel** [comment]
Very well put @usergenic. I agree on the secondary point as well.

**Comment by usergenic** [comment]
(fixed and released in 2.0.3)
